api/v1alpha1/bgppeer_types.go

```diff
@@ -0,0 +1,111 @@
+/*
+
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"time"
+)
+
+type MatchExpression struct {
+	Key      string `json:"key" yaml:"key"`
+	Operator string `json:"operator" yaml:"operator"`
+	// +kubebuilder:validation:MinItems:=1
+	Values []string `json:"values" yaml:"values"`
+}
+
+type NodeSelector struct {
```

Comment:
we can't use the same type because we customized the yaml fmt to match what configmap will have, @oribon suggested the same but I don't that will work